rpd_generator/bdl_structure/bdl_commands/system.py

```diff
@@ -1330,6 +1394,73 @@ def populate_air_energy_recovery(self):
                 self.air_energy_recovery_outdoor_airflow
             )
 
+    def populate_fan_operation_during_occupied(self):
+        fan_sch_name = self.keyword_value_pairs.get(BDL_SystemKeywords.FAN_SCHEDULE)
+
+        if (
+            self.keyword_value_pairs.get(BDL_SystemKeywords.TYPE)
+            == BDL_SystemTypes.RESVVT
+        ):
+            return FanSystemOperationOptions.CYCLING
+
+        elif (
+            self.keyword_value_pairs.get(BDL_SystemKeywords.TYPE)
+            == BDL_SystemTypes.DOAS
+        ):
+            fan_sch = self.rmd.bdl_obj_instances.get(fan_sch_name)
+            if not fan_sch:
+                return FanSystemOperationOptions.INTERMITTENT
+            else:
+                return FanSystemOperationOptions.CONTINUOUS
+
+        elif self.keyword_value_pairs.get(BDL_SystemKeywords.TYPE) in [
+            BDL_SystemTypes.PTAC,
+            BDL_SystemTypes.HP,
+            BDL_SystemTypes.UVT,
+        ]:
+            if not self.fan_sys_minimum_outdoor_airflow:
+                return FanSystemOperationOptions.CYCLING
+
+            min_oa_sch_name = self.keyword_value_pairs.get(
+                BDL_SystemKeywords.MIN_AIR_SCH
+            )
+            fan_sch = self.rmd.bdl_obj_instances.get(fan_sch_name)
+            min_oa_sch = self.rmd.bdl_obj_instances.get(min_oa_sch_name)
+            if not min_oa_sch:
+                return FanSystemOperationOptions.CONTINUOUS
+            if not fan_sch:  # (and min_oa_sch)
+                min_oa_sch_values = min_oa_sch.hourly_values
+                if min_oa_sch_values:
+                    for i in range(len(min_oa_sch_values)):
+                        if min_oa_sch_values[i] == 0:
+                            return FanSystemOperationOptions.CYCLING
+                    return FanSystemOperationOptions.CONTINUOUS
+                return FanSystemOperationOptions.CYCLING
+            if fan_sch and min_oa_sch:
+                fan_sch_values = fan_sch.hourly_values
+                min_oa_sch_values = min_oa_sch.hourly_values
+                if fan_sch_values and min_oa_sch_values:
+                    for i in range(len(fan_sch_values)):
+                        if min_oa_sch_values[i] == 0 and fan_sch_values[i] in (1, -999):
+                            return FanSystemOperationOptions.CYCLING
```

Comment:
Just documenting my tests here:

- If fan sch is all 1s or not populated and there are any OA requirements at the zone level or via a min-oa-sch then it operates continously
- If fan schedule is all 0s, regardless of OA requirements at the zone level or via min-oa-sch the fan operates according to NIGHT-CYCLE-CTRL which if set to Stay Off the fans will stay off. (I think maybe this scenario needs to be incorporated)

Reply:
hmm, I just ran a test with a fan sch of 1 and no OA and the fan operated continuously for UVT

Reply:
If fan schedule is all 0s, regardless of OA requirements at the zone level or via min-oa-sch the fan operates according to NIGHT-CYCLE-CTRL which if set to Stay Off the fans will stay off. (I think maybe this scenario needs to be incorporated)

this would be in `operation_during_unoccupied` right?

Reply:
It would apply to both because it is off 24/7 and then we do not know what the hec is going in terms of occ versus unocc so I think it would be Stay_Off or whatever the enumeration is for both.

Reply:
Ok so you think we should specifically handle the scenario where all 8,760 values in the fan schedule are 0 as special case. We can't tell what is occupied vs unoccupied so they both inherit the same enumeration based on the NIGHT-CYCLE-CTRL?

Reply:
I agree with this approach.